tmf/org.eclipse.tracecompass.tmf.core/src/org/eclipse/tracecompass/tmf/core/event/aspect/ITmfEventAspect.java

```diff
@@ -107,4 +108,16 @@ public interface ITmfEventAspect<T> {
     default boolean isHiddenByDefault() {
         return false;
     }
+
+
+    /**
+     * This method will return the data type of the aspect.
+     * The data type is considered as String by default
+     *
+     * @return the data type of the aspect
+     * @since 9.3
+     */
+    default DataType getDataType() {
```

Comment:
This method should be added also to `ISegmentAspect` interface and related implementations of that interface (See also SegmentStoreTableDataProvider that use `ISegmentAspect`)

Reply:
I investigated this suggestion, but it doesn't sound proper;

1. The `DataType getDataType()` in the `ITmfEventAspect` class returns the data type based on the `DataType` enum.
2. The `SegmentType getType()` in the `ISegmentAspect` class returns the segment type based on the `SegmentType` enum.
3. Your recommendation is to add a `DataType`-based method in the `ISegmentAspect` class to return the data type of the segment. However, the type that a segment can have is either `CATEGORICAL` or `CONTINUOUS`, which are not present in the `DataType` enum. 
4. As we need to have a consistent enum type for both classes to use in the `TmfTreeDataModel` class, your suggestion may not be suitable.

Hence, I was thinking of having a common enum interface for `DataType` and `SegmentType`, so we can handle it easier.

Reply:
I think CATEGORICAL/CONTINUOUS and the values of DataType are complementary and not just one each. A column can have DataType NUMBER and can be CONTINUOUS or a column could have a DataType NUMBER but it's CATEGORICAL. So, you can't just have one enum which includes values of DataType and SegmentType. Or do I miss understand something?

@MatthewKhouzam any comment from your side?

Reply:
Oh, I didn't know that a segment can have both types (data type + segment type) at the same time. In this case, that wouldn't be a problem, and your solution works fine. I guess @MatthewKhouzam can also verify it.